includes/Query.php

```diff
@@ -89,7 +86,15 @@ public function fetch() {
 		}
 
 		if ($this->threadMode) {
-			$this->totalCount = intval($dbr->query('select FOUND_ROWS() as row')->fetchObject()->row);
+			$builder = $dbr->newSelectQueryBuilder()
+				->select('COUNT(*) as count')
+				->from('FlowThread')
+				->where($cond)
+				->caller(__METHOD__);
+			if ($this->limit !== -1) {
+				$builder->limit($this->limit);
```

Comment:
The limit here is useless; it's limiting the total number of rows returned (which will always be 1, and if it were limiting the total number of rows queried before `COUNT(*)`, then it again returns the number of commented after limit not before it, so it's wrong either way).

PS. I haven't been following the development of MediaWiki for a while; does the query builder have any advantage over just `select`?

Reply:
[The mediawiki documentation](https://www.mediawiki.org/wiki/Manual:Database_access#SelectQueryBuilder) recommends using SelectQueryBuilder to query.